src/components/Calendar/CalenderComponent/EventModal.tsx

```diff
@@ -1,29 +1,90 @@
-    label: string;
-    day: number | null;
-    timeFrom: string;
-    timeTo: string;
-    id: string | number;
-}
+export const labelsClasses = new Map([
+    ["INTERVIEW", "green"], 
+    ["PPT", "red"], 
+    ["TEST", "indigo"], 
+    ["COMPLETED", "blue"], 
+    ["APPLICATION", "purple"]
+]);
+const colors = ["bg-green-300", "bg-red-300", "bg-indigo-300", "bg-gray-300", "bg-blue-300", "bg-purple-300"];
+const colorsHeader = ["bg-green-100", "bg-red-100", "bg-indigo-100", "bg-gray-100", "bg-blue-100", "bg-purple-100"];
 
 export default function EventModal() {
-    const { setShowEventModal, daySelected, dispatchCallEvents, selectedEvent,timeFrom,timeTo,setTimeFrom,setTimeTo } = useContext(GlobalContext);
+    const { setShowEventModal, daySelected, selectedEvent,addEvent,updateEvent,deleteEvent} = useContext(GlobalContext);
 
-    const [title, setTitle] = useState<string>(selectedEvent ? selectedEvent.title : "");
-    const [description, setDescription] = useState<string>(selectedEvent ? selectedEvent.description : "");
-    const [selectedLabel, setSelectedLabel] = useState<string>(selectedEvent ? labelsClasses.find(lbl => lbl === selectedEvent.label) || labelsClasses[0] : labelsClasses[0]);
+    const [title, setTitle] = useState<string>(selectedEvent ? selectedEvent.job.company.name : "");
+    const [description, setDescription] = useState<string>(selectedEvent ? selectedEvent.metadata : "");
+    const [selectedLabel, setSelectedLabel] = useState<[string, string]>(selectedEvent ? [selectedEvent.type, labelsClasses.get(selectedEvent.type) || ""] : ["", ""]);    
+    // const [selectedLabel, setSelectedLabel] = useState<string>(selectedEvent ? labelsClasses.find(lbl => lbl === selectedEvent.type) || labelsClasses[0] : labelsClasses[0]);
     const [timeSelector, setTimeSelector] = useState<boolean>(false);
     const [showTimeList, setShowTimeList] = useState<{ from: boolean, to: boolean }>({ from: false, to: false });
+    const [timeFrom,setTimeFrom] = useState(selectedEvent ? dayjs(selectedEvent.startDateTime).format('hh:00 A') : "from")
+    const [timeTo,setTimeTo] = useState(selectedEvent ? dayjs(selectedEvent.endDateTime).format('hh:00 A') : "to")
+    const [checked,setChecked] = useState(selectedEvent ? selectedEvent.visibleToRecruiter : false)
+    // const [type,setType] = useState(selectedEvent ? selectedEvent.type : "")
+    const [companyName,setCompanyName] = useState(selectedEvent ? selectedEvent.job.company.name : "")
+    const [role,setRole] = useState(selectedEvent ? selectedEvent.job.role : "")
+    const [recruitmentType,setRecruitmentType] = useState(selectedEvent ? selectedEvent.job.season.type : "")
+    const [roundNumber,setRoundNumber] = useState(selectedEvent ? selectedEvent.roundNumber : 1)
+    const[IsOpen,setIsOpen] = useState<boolean>(false)
+
+    function displayColor(label:string){
+        return labelsClasses.get(label);
+      }
+
+    function convertToDateTime(timeString, dayjsObject) {
```

Comment:
there might be a JS libarary for it. please see if we can use that. the code is prone to errors if we do this ourselves